src/Compilers/CSharp/Portable/Errors/ErrorCode.cs

```diff
@@ -1232,17 +1232,20 @@ internal enum ErrorCode
         ERR_FixedBufferTooManyDimensions = 7092,
         ERR_CantReadConfigFile = 7093,
         ERR_BadAwaitInCatchFilter = 7094,
-        WRN_FilterIsConstant = 7095,
+        // WRN_FilterIsConstant = 7095,   // replaced by FilterIsConstantFalse and FilterIsConstantTrue
         ERR_EncNoPIAReference = 7096,
         //ERR_EncNoDynamicOperation = 7097,   // dynamic operations are now allowed
         ERR_LinkedNetmoduleMetadataMustProvideFullPEImage = 7098,
         ERR_MetadataReferencesNotSupported = 7099,
         ERR_InvalidAssemblyCulture = 7100,
         ERR_EncReferenceToAddedMember = 7101,
         ERR_MutuallyExclusiveOptions = 7102,
+        WRN_FilterIsConstantFalse = 7103,
+        WRN_FilterIsConstantTrue = 7104,
+        WRN_FilterIsConstantRedundantTryCatch = 7105,
```

Comment:
Thank you, @sharwell . 
Should I then just distinguish True and False cases, and suggest `consider removing the catch clause` for all False cases? I agree it would reduce unnecessary overhead for the two false cases. 
Or should this be discussed with @gafter who originally raised #18750 ?